vortex-filtering/include/vortex_filtering/probability/multi_var_gauss.hpp

```diff
@@ -0,0 +1,119 @@
+#pragma once
+#include <Eigen/Dense>
+#include <random>
+
+namespace vortex {
+namespace prob {
+
+/**
+ * A class for representing a multivariate Gaussian distribution
+ * @tparam N_DIMS dimentions of the Gaussian
+ */
+template <size_t n_dims> class MultiVarGauss {
+public:
+  static constexpr int N_DIMS = (int)n_dims;
```

Comment:
What's the reason for templating a `size_t` but casting it to `int` here? (i.e. could N_DIMS just be size_t as well?)

Reply:
The Eigen types needs int (to allow for the -1 for dynamic sizes). Won't compile otherwise. I found it easier to cast it once rather than for every Eigen type I define